proposed/vector-instruction-set-default.md

```diff
@@ -0,0 +1,100 @@
+
+At the same time, we need to make choices that move .NET metrics and capabilities forward. Let's explore this topic.
+
+Supporting ecosystem data:
+
+- [Steam hardware survey](https://store.steampowered.com/hwsurvey) (see "Other Settings")
+- [Azure VMs](https://azure.microsoft.com/en-us/pricing/details/virtual-machines/series/) -- these all support AVX2; we expect same for other clouds
+
+Note: "SSEx" is used to refer to any of the SSE versions, like SSE2 and SSE4.
+
+## .NET 5.0 behavior
+
+Let's start with the SIMD characteristics for .NET 5.0:
+
+- Ready-to-run images target SSE2 instructions for vectorized code (not just by default; it's the only option). That means ready-to-run images  include processor-specific machine code with SSE2 instructions and never any AVX/AVX2 instructions.
+- The JIT (via tiered compilation) can re-jit methods so that they take advantage of a newer SIMD instruction set (if present on a given machine), like SSE4, AVX, or AVX2. The JIT doesn't do this proactively, but will only rejit methods that are used a lot, per its normal behavior. It doesn't re-jit methods solely to use better SIMD instructions, but will use the best SIMD instructions available (on a given machine) when it does rejit methods, again, per its normal behavior.
+- As a result, applications rarely run at peak performance initially (since most machines support newer SIMD instructions than SSE2) and it takes a long time for an app to get to peak performance (via tiering to SSE4, AVX, or AVX2-enabled code).
+- This behavior is fine for machines with only SSE2, and is leaving performance on the table for newer machines.
+
+## Context, characteristics and constraints
+
+The most obvious solution is to target AVX2 by default, and this approach would be the clear winner for anyone using a machine that was bought after 2015. The challenge is that this change would be significant regression for SSEx machines, even more significant than the win it would provide for AVX2-capable machines, in the general case.
+
+The primary issue is that ready to run code for (what is likely to be) performance sensitive methods would no longer be usable on SSEx machines, but would require JITing IL methods instead. The resultant native code would be the same (actually, it would be better; we'll leave that topic for another day), but would require time to produce via JIT compilation. As a result, startup time for apps on SSEx machines would be noticeably degraded. We guess the startup regression would be unacceptable, but need to measure it.
+
+Making AVX2 the default would be the same as dropping support for SSEx. If we do that, we'd need to announce it, and reason about what an SSEx offering looks like, if any.
```

Comment:
Is this the same as dropping support or just that it may cause more initial jitting for hardware without AVX2 support?

That is, if an image is R2R generated for AVX2, but the hardware doesn't support AVX2, would the app not be runnable or would it just have additional startup overhead because it would have to treat the R2R code as invalid?

Reply:
> would the app not be runnable or would it just have additional startup overhead because it would have to treat the R2R code as invalid?

Just slower. We don't plan a failure case is this scenario.

> Is this the same as dropping support or just that it may cause more initial jitting for hardware without AVX2 support?

We haven't done performance measurements yet. Once we have those, it will be easier to answer your question. The size of the cost is really the answer.

I had the wording "in essence" earlier. I should add it back. That better describes what I was trying to say. If the performance regression is high, then the experience will be unusable, which is "in essence" the same as dropping support. We'll see.